tests/ci/integration/run_mariadb_integration.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash -exu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+source tests/ci/common_posix_setup.sh
+
+trap dump_log EXIT
+
+# Set up environment.
+
+# ROOT
+#  |
+#  - AWS_LC_DIR
+#    |
+#    - aws-lc
+#  |
+#  - SCRATCH_FOLDER
+#    |
+#    - mariadb
+#    - AWS_LC_BUILD_FOLDER
+#    - AWS_LC_INSTALL_FOLDER
+#    - MARIADB_BUILD_FOLDER
+
+# The CFlags, skipped test list, and patches were taken from the internal CI pipeline of our customer.
+
+# Assumes script is executed from the root of aws-lc directory
+AWS_LC_DIR=$(pwd)
+cd ../
+ROOT=$(pwd)
+
+SCRATCH_FOLDER=${ROOT}/"MARIADB_BUILD_ROOT"
+MARIADB_SRC_FOLDER="${SCRATCH_FOLDER}/server"
+MARIADB_BUILD_FOLDER="${SCRATCH_FOLDER}/server/mariadb-aws-lc"
+MARIADB_PATCH_FOLDER=${AWS_LC_DIR}/"tests/ci/integration/mariadb_patch"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${MARIADB_SRC_FOLDER}/aws-lc-install"
+
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf ${SCRATCH_FOLDER}/*
+cd ${SCRATCH_FOLDER}
+
+function mariadb_build() {
+  cmake ${MARIADB_SRC_FOLDER} -GNinja -DWITH_SSL=${AWS_LC_INSTALL_FOLDER} "-B${MARIADB_BUILD_FOLDER}" -DPLUGIN_COLUMNSTORE=NO -DPLUGIN_ROCKSDB=NO -DPLUGIN_S3=NO -DPLUGIN_MROONGA=NO -DPLUGIN_CONNECT=NO -DPLUGIN_SPHINX=NO -DPLUGIN_SPIDER=NO -DPLUGIN_TOKUDB=NO -DPLUGIN_PERFSCHEMA=NO -DWITH_WSREP=OFF
+  ninja -C ${MARIADB_BUILD_FOLDER}
+  ls -R ${MARIADB_BUILD_FOLDER}
+}
+
+# Used to access debugging logs.
+function dump_log() {
+  ls ${MARIADB_BUILD_FOLDER}/mysql-test/var/log
+  for logfile in $(find -L "${MARIADB_BUILD_FOLDER}/mysql-test/var/log" -type f -name '*.log'); do
+    echo "Dumping out logs to observe:"
+    cat $logfile
+  done
+}
+
+function mariadb_run_tests() {
+  pushd ${MARIADB_BUILD_FOLDER}
+  # More complicated integration tests. mtr expects to be launched in-place and with write access to it's own directories
+  #
+  # main.plugin_load passes, but is skipped over since it generates a warning when we run the script in Codebuild. Warnings will cause
+  # a failure in MariaDB's test runs, unless --nowarnings is turned on. The warning is not reproducable in Gitlab's CI or any local
+  # container runs. This test isn't relevant to AWS-LC integration so we skip over the Codebuild specific issue for now.
+  pushd mysql-test
```

Comment:
```suggestion
  pushd ${MARIADB_BUILD_FOLDER}/mysql-test
  # More complicated integration tests. mtr expects to be launched in-place and with write access to it's own directories
  #
  # main.plugin_load passes, but is skipped over since it generates a warning when we run the script in Codebuild. Warnings will cause
  # a failure in MariaDB's test runs, unless --nowarnings is turned on. The warning is not reproducable in Gitlab's CI or any local
  # container runs. This test isn't relevant to AWS-LC integration so we skip over the Codebuild specific issue for now.
```